src/convertToObject.js

```diff
@@ -6,7 +6,15 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  return sourceString
+    .trim()
+    .split(';')
+    .filter(Boolean)
+    .reduce((acc, rule) => {
+      const [property, value] = rule.split(':').map((part) => part.trim());
+
+      return { ...acc, [property]: value };
```

Comment:
[CODE KNOWLEDGE] - The use of the spread operator in the reduce callback (line 16) to copy the accumulator object on each iteration is unnecessary and can lead to performance issues, especially with a large number of CSS rules. Consider using Object.assign or directly assigning the new property to the accumulator object.